pytest_asyncio/plugin.py

```diff
@@ -145,10 +145,17 @@ def pytest_pyfunc_call(pyfuncitem):
     function call.
     """
     for marker_name, fixture_name in _markers_2_fixtures.items():
-        if marker_name in pyfuncitem.keywords \
-                and not getattr(pyfuncitem.obj, 'is_hypothesis_test', False):
+        if marker_name in pyfuncitem.keywords:
             event_loop = pyfuncitem.funcargs[fixture_name]
-
+            if getattr(pyfuncitem.obj, 'is_hypothesis_test', False):
+                # Wrap the async function to execute in the pytest-asyncio event loop and yield to delegate
+                # test execution to the hypothesis plugin
+                pyfuncitem.obj.hypothesis.inner_test = wrap_in_sync(
+                    pyfuncitem.obj.hypothesis.inner_test, event_loop
+                )
+                # It is important to yield, not return. Otherwise the event_loop fixture will get cleaned up too early
+                # when using a combination of hypothesis and pytest.mark.parametrize
+                yield None
```

Comment:
As the comment says, `return None` will cause problems with parametrized hypothesis tests. However, changing `return None` to `yield None` breaks `test_simple::test_asyncio_marker_fail` for no apparent reason. Do you have any idea why this might cause `xfail` to break?